ci/travis/build-docker-images.py

```diff
@@ -0,0 +1,230 @@
+import datetime
+import glob
+import os
+import re
+import shutil
+import subprocess
+import sys
+from typing import List
+
+import docker
+
+DOCKER_USERNAME = "raytravisbot"
+DOCKER_CLIENT = None
+
+
+def _subprocess_wrapper(cmd):
+    result = subprocess.run(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    if result.returncode != 0:
+        print(f"COMMAND FAILED: {cmd}")
+        print(f"STDOUT is: {result.stdout}")
+        print(f"STDERR is: {result.stderr}")
+        raise Exception("Command Failed") from None
+    return (result.stdout or b"").decode()
+
+
+def _merge_build():
+    return os.environ.get("TRAVIS_PULL_REQUEST") == "false"
+
+
+def _release_build():
+    branch = os.environ.get("$TRAVIS_BRANCH")
+    if not branch:
+        print("Branch not found!")
+        return False
+    return branch != "master" and "releases" in branch
+
+
+def _get_curr_dir():
+    return os.path.dirname(os.path.realpath(__file__))
+
+
+def _get_root_dir():
+    return os.path.join(_get_curr_dir(), "../../")
+
+
+def _get_wheel_name():
+    matches = glob.glob(f"{_get_root_dir()}/.whl/*cp37m-manylinux*")
+    assert len(matches) == 1
+    f"Found ({len(matches)}) matches '*cp37m-manylinux*' instead of 1"
+    return os.path.basename(matches[0])
+
+
+def _docker_affected():
+    variable_definitions = _subprocess_wrapper(
+        f"python {_get_curr_dir()}/determine_tests_to_run.py").split()
+    env_var_dict = {
+        x.split("=")[0]: x.split("=")[1]
+        for x in variable_definitions
+    }
+    affected = env_var_dict["RAY_CI_DOCKER_AFFECTED"] == "1" or \
+        env_var_dict["RAY_CI_PYTHON_DEPENDENCIES_AFFECTED"] == "1"
+    print(f"Docker affected: {affected}")
+    return affected
+
+
+def _build_helper(image_name) -> List[str]:
```

Comment:
_build_cpu_gpu_images
🤖 naming is hard!

Reply:
lol--great idea!